pkg/k8s/apis/cilium.io/v1alpha1/types.go

```diff
@@ -55,7 +55,7 @@ type KProbeArg struct {
 	// +kubebuilder:validation:Minimum=0
 	// Position of the argument.
 	Index uint32 `json:"index"`
-	// +kubebuilder:validation:Enum=auto;int;int8;uint8;int16;uint16;uint32;int32;uint64;int64;char_buf;char_iovec;size_t;skb;sock;string;fd;file;filename;path;nop;bpf_attr;perf_event;bpf_map;user_namespace;capability;kiocb;iov_iter;cred;load_info;module;syscall64;kernel_cap_t;cap_inheritable;cap_permitted;cap_effective;linux_binprm;data_loc;net_device
+	// +kubebuilder:validation:Enum=auto;int;int8;uint8;int16;uint16;uint32;int32;uint64;int64;char_buf;char_iovec;size_t;skb;sock;string;fd;file;filename;path;permission;nop;bpf_attr;perf_event;bpf_map;user_namespace;capability;kiocb;iov_iter;cred;load_info;module;syscall64;kernel_cap_t;cap_inheritable;cap_permitted;cap_effective;linux_binprm;data_loc;net_device
```

Comment:
I believe that we don't need that change. Let me provide my understanding and correct me if I am wrong. An example tracing policy for your PR is:
```
apiVersion: cilium.io/v1alpha1
kind: TracingPolicy
metadata:
  name: "path-chmod"
spec:
  kprobes:
  - call: "security_path_chmod"
    syscall: false
    args:
    - index: 0
      type: "path"
    - index: 1
      type: "uint16"
```
So what you implemented here adds a new permission field if the user defines a type which is "path", "file", or some other values. But the user cannot specify `type: "permission"`. This is the list of possible types, so you can remove `permission` from that list (and all autogenerated code).

Reply:
Thanks for your comments,  I have update  this PR. Pls take a look.